### PR TITLE
Feature/objectid and decimal128

### LINKED ISSUE #2674: Decimal128 data type
See RPM-88 for details.

* [x] Add `bson` as dependency
* [x] Make `bson.Decimal128` constructor accessible from C++
* [x] Accessors (node.js)
* [x] Accessors (RN)
* [x] Schema parsing and normalization
* [x] Testing: create and update object
* [x] Testing: queries

PR: https://github.com/realm/realm-js/pull/2724

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #2674  and #2675 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: src/js_realm.cpp [hunk 1]
```diff
@@ -103,6 +103,12 @@ std::string TypeErrorException::type_string(Property const& prop)
         case PropertyType::Data:
             ret = "binary";
             break;
+        case PropertyType::Decimal:
+            ret = "decimal128";
```
Comment (bmunkholm) [inline src/js_realm.cpp]: Should it be "decimal"?

File: src/js_realm.cpp [hunk 2]
```diff
@@ -103,6 +103,12 @@ std::string TypeErrorException::type_string(Property const& prop)
...
+            break;
+        case PropertyType::ObjectId:
+            ret = "ObjectId";
```
Comment (bmunkholm) [inline src/js_realm.cpp]: should it be  "object id" as other places

File: src/jsc/jsc_value.hpp [hunk 3]
```diff
@@ -125,12 +125,12 @@ inline bool jsc::Value::is_valid(const JSValueRef &value) {
 
 template<>
 inline bool jsc::Value::is_decimal128(JSContextRef ctx, const JSValueRef &value) {
-    return true; // FIXME: must be an instance of Decimal128 from BSON
+    return JSValueIsObject(ctx, value); // FIXME: can we do better?
```
Comment (blagoev) [inline src/jsc/jsc_value.hpp]: if it is `Decimal128` it will have the property named `_bsontype =  "Decimal128"`

Comment (kraenhansen) [reply src/jsc/jsc_value.hpp]: Perhaps https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/isPrototypeOf could be useful here?

**Comment by kneth** [comment]
Succeeded by #2724

**Pull request closed without merging**